components/feature/intent/src/main/java/mozilla/components/feature/intent/IntentProcessor.kt

```diff
@@ -67,15 +67,16 @@ class IntentProcessor(
         when {
             TextUtils.isEmpty(extraText.trim()) -> false
 
-            extraText.isUrl() -> {
-                val session = createSession(extraText, source = Source.ACTION_SEND)
-                sessionUseCases.loadUrl.invoke(extraText, session)
-                true
-            }
             else -> {
-                val session = createSession(extraText, source = Source.ACTION_SEND)
-                searchUseCases.defaultSearch.invoke(extraText, session)
-                true
+                val url = extraText.split(" ").find { it.isUrl() }
```

Comment:
Note that `isUrl()` is a very basic implementation to only determine whether something entered into the URL bar should trigger a search or a load. I doesn't actually look for somewhat valid URLs. I fear that it will do unexpected things. For example if I share the text `Hello world.` then from the implementation of `isUrl()` I guess it would try to load `world.`.

Fennec has a more complex `WebURLFinder` for that:
* https://searchfox.org/mozilla-central/source/mobile/android/services/src/main/java/org/mozilla/gecko/sync/setup/activities/WebURLFinder.java
* https://searchfox.org/mozilla-central/source/mobile/android/services/src/test/java/org/mozilla/gecko/sync/setup/activities/WebURLFinderTest.java

Not something we would need to fix now though; we could also look at landing something like `WebURLFinder` in a follow-up.

Reply:
OK, I've filed a follow-up: https://github.com/mozilla-mobile/android-components/issues/1438